agent/internal/nvidia.go

```diff
@@ -43,17 +45,69 @@ func getNvidiaVersion() (string, error) {
 	return record[0], nil
 }
 
-func nvidiaRuntimeInstalled() (bool, error) {
-	c, err := client.NewClientWithOpts(client.FromEnv)
+// dockerCanUseGPUs determines whether Docker can run containers with GPUs. Unfortunately, it seems
```

Comment:
This seems pretty unfortunate to have to do this -- e.g., potentially requires pulling an extra container image, which might not work in air-gapped settings, and will slow startup time. I suppose there's no better way to do this? I wonder if we should try filing something with upstream to see if they can either improve this or tell us if there's a better way to do it?

Reply:
Okay, I'll check upstream.